osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/DlmsConnectionHelper.java

```diff
@@ -84,6 +89,10 @@ public void createAndHandleConnectionForDevice(
       final Consumer<DlmsConnectionManager> taskForConnectionManager)
       throws OsgpException {
 
+    if (device.getIpAddress() == null) {
+      this.domainHelperService.setIpAddressFromMessageMetadataOrSessionProvider(
+          device, messageMetadata);
```

Comment:
setIpAddressFromMessageMetadataOrSessionProvider will now be invoked twice...... Maybe we should move the if == null to the method setIpAddressFromMessageMetadataOrSessionProvider

Reply:
The null check is moved to the method and a unit test is added.